### PR TITLE
[r+] Intpocalypse, book edition.

### PR DESCRIPTION
**Pull request by steveklabnik**
Fix all usage of int/uint/i/u in the book.

r? @nikomatsakis  who called this to my attention.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
with respect to the choice of reviewer: No good deed goes unpunished.

**Comment by steveklabnik** [comment]
@nikomatsakis :heart: :sweat_smile:

File: src/doc/trpl/generics.md [hunk 1]
```diff
@@ -40,26 +40,26 @@ we substitute that type for the same type used in the generic. Here's an
...
+the right-hand side of the binding, we do make a `Some(T)`, where `T` is `5`.
+Since that's an `i32`, the two sides match, and Rust is happy. If they didn't
 match, we'd get an error:
 
 ```{rust,ignore}
-let x: Option<f64> = Some(5i);
+let x: Option<f64> = Some(5);
 // error: mismatched types: expected `core::option::Option<f64>`
-// but found `core::option::Option<int>` (expected f64 but found int)
+// but found `core::option::Option<i32>` (expected f64 but found i32)
```
Comment (tbu-) [inline src/doc/trpl/generics.md]: This should be
```error: mismatched types: expected `core::option::Option<f64>`, found `core::option::Option<_>` (expected f64, found integral variable)```

File: src/doc/trpl/unsafe.md [hunk 2]
```diff
@@ -667,24 +667,24 @@ extern {
...
 #[lang="exchange_free"]
-unsafe fn deallocate(ptr: *mut u8, _size: uint, _align: uint) {
+unsafe fn deallocate(ptr: *mut u8, _size: usize, _align: usize) {
     libc::free(ptr as *mut libc::c_void)
 }
 
 #[start]
-fn main(argc: int, argv: *const *const u8) -> int {
-    let x = box 1i;
+fn main(argc: isize, argv: *const *const u8) -> isize {
```
Comment (tbu-) [inline src/doc/trpl/unsafe.md]: I believe this is wrong, I'm not sure though.

My suggestion:
```fn main(argc: c_int, argv: *const *const u8) -> c_int;```

Comment (steveklabnik) [inline src/doc/trpl/unsafe.md]: The compiler told me this was the signature.

Comment (nikomatsakis) [inline src/doc/trpl/unsafe.md]: It does seem surprising. I think there was an open bug on this somewhere. cc @alexcrichton

Comment (tbu-) [inline src/doc/trpl/unsafe.md]: The open bug is https://github.com/rust-lang/rust/issues/20064.

File: src/doc/trpl/pointers.md [hunk 3]
```diff
@@ -544,12 +546,17 @@ for more detail on how lifetimes work.
 Using boxes and references together is very common. For example:
 
 ```{rust}
+<<<<<<< HEAD
 fn add_one(x: &int) -> int {
+=======
```
Comment (goffrie) [inline src/doc/trpl/pointers.md]: Conflict markers? :fearful:

Comment (steveklabnik) [inline src/doc/trpl/pointers.md]: ffuuuu

**Comment by steveklabnik** [comment]
This made it in as part of my rolllup

**Pull request closed without merging**